### PR TITLE
feat: add support outputting rekor response on signing

### PR DESCRIPTION
**Pull request by vishal-chdhry**
#### Summary

Closes: #3110

Previously, there's no support for outputting a bundle with `cosign sign` ([code](https://github.com/sigstore/cosign/blob/main/cmd/cosign/cli/sign/sign.go#L266-L298)), only `sign-blob` ([code](https://github.com/sigstore/cosign/blob/main/cmd/cosign/cli/sign/sign_blob.go#L135-L152)). This PR adds support for outputting rekor response on signing. This supports use-cases where a signer does not want to attach metadata immediately to the container.

#### Release Note

Added support outputting rekor response on signing with `cosign sign`

### CODE REVIEW

File: cmd/cosign/cli/sign/sign.go [hunk 1]
```diff
@@ -591,3 +617,33 @@ func (c *SignerVerifier) Bytes(ctx context.Context) ([]byte, error) {
 	}
 	return pemBytes, nil
 }
+
+func fetchSignedPayload(sig oci.Signature) (*cosign.SignedPayload, error) {
```
Comment (haydentherapper) [inline cmd/cosign/cli/sign/sign.go:621]: Can this return a `LocalSignedPayload` instead to be in sync with sign-blob? It should be loadable with https://github.com/sigstore/cosign/blob/main/pkg/cosign/fetch.go#L199.

**Review by haydentherapper — CHANGES_REQUESTED** [review]
Can you add a test for this too?

**Comment by vishal-chdhry** [comment]
@haydentherapper I have added a test and updated the logic, can you please review it

Besides, I cannot run/debug tests in `test/e2e_test.go`. I think `gopls` is not working for me in that file, autocomplete does not work either. Is there anything I need to configure? I am using VSCode.

File: test/e2e_test.go [hunk 2]
```diff
@@ -951,6 +951,51 @@ func TestRekorBundle(t *testing.T) {
 	must(verify(pubKeyPath, imgName, true, nil, ""), t)
 }
 
+func TestRekorOutput(t *testing.T) {
+	// turn on the tlog
+	defer setenv(t, env.VariableExperimental.String(), "1")()
```
Comment (haydentherapper) [inline test/e2e_test.go:956]: Can you remove, the experimental flag is no longer used?

**Review by haydentherapper — DISMISSED** [review]
One tiny comment, looks good!

**Comment by vishal-chdhry** [comment]
@haydentherapper Removed the flag

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**